flang/lib/Lower/Bridge.cpp

```diff
@@ -1585,7 +1585,8 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     if (unstructuredContext) {
       // When transitioning from unstructured to structured code,
       // the structured code might be a target that starts a new block.
-      maybeStartBlock(eval.isConstruct() && eval.lowerAsStructured()
+      maybeStartBlock(eval.isConstruct() && eval.lowerAsStructured() &&
```

Comment:
There's a different idiom for doing this check in include/flang/Lower/PFTBuilder.h

```
    if (evaluationList.size())
      return evaluationList.front().position;
```

Reply:
Oh. I'll look for it. It should use empty(), since empty() is constant time and size() in linear time per this data structure.